plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4672,22 +4714,64 @@ public void adjust(Mechanism mechanism) {
         // @tags
         // <LocationTag.sign_contents>
         // -->
-        if (mechanism.matches("sign_contents") && getBlockState() instanceof Sign) {
-            Sign state = (Sign) getBlockState();
-            for (int i = 0; i < 4; i++) {
-                PaperAPITools.instance.setSignLine(state, i, "");
-            }
-            ListTag list = mechanism.valueAsType(ListTag.class);
-            CoreUtilities.fixNewLinesToListSeparation(list);
-            if (list.size() > 4) {
-                mechanism.echoError("Sign can only hold four lines!");
+        if (mechanism.matches("sign_contents")) {
+            BlockState state = getBlockState();
+            if (!(state instanceof Sign sign)) {
+                mechanism.echoError("'sign_contents' mechanism can only be called on Sign blocks.");
+            } else {
```

Comment:
If you want to update this mechanism should move it to modern `tagProcessor#registerMechanism`.
Also should use early return for these error checks instead of nested if/else